### PR TITLE
apk: Add no-cache option

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Add `--no-cache` option to apk module.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
apk

### CODE REVIEW

**Comment by aminvakil** [comment]
https://app.shippable.com/github/ansible-collections/community.general/runs/2285/1/tests

I cannot understand why EXAMPLES is not a valid yaml file.

**Comment by aminvakil** [comment]
Other failures should be fixed by merging https://github.com/ansible-collections/community.general/pull/549

File: CHANGELOG.rst [hunk 1]
```diff
@@ -40,6 +40,7 @@ Minor Changes
 - ali_instance_info - Add params ``name_prefix``, ``filters``
 - alicloud modules - Add authentication params to all modules
 - alicloud modules - now only support Python 3.6, not support Python 2.x
+- apk - add no-cache option
```
Comment (felixfontein) [inline CHANGELOG.rst:43]:
```suggestion
```
Do not touch this file. It is automatically generated.

File: plugins/modules/packaging/os/apk.py [hunk 2]
```diff
@@ -30,6 +30,11 @@
       - A package name, like C(foo), or multiple packages, like C(foo, bar).
     type: list
     elements: str
+  no_cache:
+    description:
+      - Do not use any local cache path
+    type: bool
+    default: 'no'
```
Comment (felixfontein) [inline plugins/modules/packaging/os/apk.py:37]:
```suggestion
    default: 'no'
    version_added: 1.0.0
```

File: plugins/modules/packaging/os/apk.py [hunk 3]
```diff
@@ -30,6 +30,11 @@
       - A package name, like C(foo), or multiple packages, like C(foo, bar).
     type: list
     elements: str
+  no_cache:
+    description:
+      - Do not use any local cache path
```
Comment (felixfontein) [inline plugins/modules/packaging/os/apk.py:35]:
```suggestion
      - Do not use any local cache path.
```

File: plugins/modules/packaging/os/apk.py [hunk 4]
```diff
@@ -119,6 +124,12 @@
     state: latest
     update_cache: yes
     repository: http://dl-3.alpinelinux.org/alpine/edge/main
+
+- name: Install package without using cache and caching repositories
+  apk:
+    name: foo
+    state:latest
```
Comment (felixfontein) [inline plugins/modules/packaging/os/apk.py:131]:
```suggestion
    state: latest
```

File: plugins/modules/packaging/os/apk.py [hunk 5]
```diff
@@ -294,6 +305,7 @@ def main():
             state=dict(default='present', choices=['present', 'installed', 'absent', 'removed', 'latest']),
             name=dict(type='list', elements='str'),
             repository=dict(type='list'),
+            no_cache=dict(default='no', type='bool'),
```
Comment (felixfontein) [inline plugins/modules/packaging/os/apk.py:308]:
```suggestion
            no_cache=dict(default=False, type='bool'),
```
(The other entries with `default='no'` are also wrong and should be fixed eventually - though not in this PR.)

File: changelogs/fragments/548_apk.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- apk - add no-cache option
```
Comment (felixfontein) [inline changelogs/fragments/548_apk.yml:2]:
```suggestion
- apk - added ``no_cache`` option (https://github.com/ansible-collections/community.general/pull/548).
```

File: plugins/modules/packaging/os/apk.py (hunk 4 above)
Comment (aminvakil) [reply plugins/modules/packaging/os/apk.py]: Oops. Thanks!

**Comment by felixfontein** [comment]
Ah, unfortunately now you have to rebase this PR after #554 got merged.

**Comment by aminvakil** [comment]
The rebase wasn't successful and I removed all commits for this PR by mistake, edited this again, could you please review it once again in case I haven't missed something?

**Comment by felixfontein** [comment]
@aminvakil thanks for implementing this!

**Comment by tdtrask** [comment]
@aminvakil Thanks for your contribution, and @felixfontein Thanks for your review. You worked so fast I never even had a chance to review the change. :)

**Comment by felixfontein** [comment]
@tdtrask this one looked safe enough ;) Please still take a look whether it looks good to you :)

**Comment by tdtrask** [comment]
@felixfontein I looked, and it's all good. Thanks.

**Pull request merged**